homework/index.js

```diff
@@ -29,16 +32,62 @@
     });
     return elem;
   }
+  function app(x) {
+    x.sort((a, b) => a.name.localeCompare(b.name));
+    const root = document.getElementById('root');
+    const select = createAndAppend('select', root, { id: 'select' });
+    for (let i = 0; i < x.length; i++) {
+      createAndAppend('option', select, { text: x[i].name, value: i });
+    }
+    const div1 = createAndAppend('div', root, { class: 'tired' });
+    const list = createAndAppend('ul', div1, { id: 'ul' });
+    const rep = createAndAppend('li', list);
+    const des = createAndAppend('li', list);
+    const fork = createAndAppend('li', list);
+    const up = createAndAppend('li', list);
 
-  function main(url) {
-    fetchJSON(url, (err, data) => {
-      const root = document.getElementById('root');
-      if (err) {
-        createAndAppend('div', root, { text: err.message, class: 'alert-error' });
-      } else {
-        createAndAppend('pre', root, { text: JSON.stringify(data, null, 2) });
+    const list2 = createAndAppend('ul', div1, { text: '' });
+    function renderContributer(cont1) {
+      for (let i = 0; i < cont1.length; i++) {
+        const link = createAndAppend('a', list2, {
+          target: ' _blank',
+          href: `${cont1[i].html_url}`,
+        });
+        const list3 = createAndAppend('li', link, {
+          text: `${cont1[i].login}  ${cont1[i].contributions} `,
+          // createAndAppend('img' , list, href: "avatar_url")
+        });
+        createAndAppend('br', list3);
+        const img = cont1[i].avatar_url;
+        createAndAppend('img', list3, { src: img });
       }
-    });
+    }
+    function innerText() {
+      des.innerText = `Description: ${x[select.value].description}`;
+      if (x[select.value].description === null) {
```

Comment:
you could have these four lines with the `createAndAppend` line making them five, in only three lines as the following:
```javascript
createAndAppend('li', list, {
  text: `Description: ${x[select.value].description}`,
  style: x[select.value].description ? 'display: block' : 'display: none'
})
```

Same as some of the next few lines.